rosetta/retriever/retriever.go

```diff
@@ -387,3 +388,50 @@ func (r *Retriever) operations(txID flow.Identifier, events []flow.Event) ([]*ob
 
 	return ops, nil
 }
+
+func (r *Retriever) AccountSequenceNumber(rosAccountID identifier.Account, keyIndex int) (uint64, error) {
+
+	// Run validation on the account qualifier. If it is valid, this will return
+	// the associated Flow account address.
+	address, err := r.validate.Account(rosAccountID)
+	if err != nil {
+		return 0, fmt.Errorf("could not validate account: %w", err)
+	}
+
+	// Get the latest block.
+	last, err := r.index.Last()
+	if err != nil {
+		return 0, fmt.Errorf("could not find last indexed block: %w", err)
+	}
+
+	// Retrieve the account at the latest block.
+	account, err := r.invoke.GetAccount(address, last)
+	if err != nil {
+		return 0, fmt.Errorf("could not retrieve account: %w", err)
+	}
+
+	// Lookup the specified key.
+	for _, key := range account.Keys {
```

Comment:
Are you sure this is needed and they are not already sorted by index? If that's the case, I would much prefer creating a lookup map for the keys by index first, and then checking just the index we want, rather than iterating through all of them. Yes, it's a bit less efficient, but that's not where we will save relevant performance. The code will be nicer, though.

Reply:
I'm not sure that they aren't already sorted. Still, the fact that there's a field titled `Index` in the structure made me feel uneasy about making any assumptions, and I thought it's best to be explicit in checking.

I've changed the code to use a lookup map now, I think the function reads more direct now.